### PR TITLE
Issue 277 recursively resolving dependencies

### PR DESCRIPTION
**Pull request by neoneo**
I've opted for passing the external beanfactory into Taffy's factory, and let getBean() ultimately check the external factory for the bean. If you have any remarks about the changes, please let me know.

### CODE REVIEW

File: core/factory.cfc [hunk 1]
```diff
@@ -41,8 +46,16 @@
...
+		<cfscript>
+			return structKeyExists(this.beans, arguments.beanName) or (arguments.includeTransients and transientExists(arguments.beanName)) or
+				(arguments.includeExternal and arguments.includeExternal and externalBeanExists(arguments.beanName));
+		</cfscript>
+	</cffunction>
+	<cffunction name="externalBeanExists" access="private" output="false" returnType="boolean">
+		<cfargument required="true" name="beanName">
 		<cfscript>
-			return structKeyExists(this.beans, arguments.beanName) or (includeTransients and transientExists(arguments.beanName));
+			return !isNull(this.externalBeanFactory) and this.externalBeanFactory.containsBean(arguments.beanName);
```
Comment (atuttle) [inline core/factory.cfc]: `isNull()` isn't available in CF8 and as of now we still support back that far.

**Comment by atuttle** [comment]
I really like your approach of passing the external factory into Taffy's factory -- that makes a heck of a lot of sense! Just one note (see the line comment) about compatibility. This is going to be a great addition. Thanks so much!

**Comment by neoneo** [comment]
Thanks for your feedback! Do you want me to fix those isNull's? Will do that later today.

**Comment by atuttle** [comment]
Yes please! Just make the appropriate changes and push to the same branch of your repo. That will automatically update this PR.

**Comment by neoneo** [comment]
I replaced the isNull calls with calls to structKeyExists (I like that better than isDefined), plus one or two minor changes (one of those is also a CF8 compat issue I believe).

**Comment by atuttle** [comment]
Awesome. I'm going to go ahead and merge it in so that it can be in the bleeding edge downloads, but it won't be included in an official release until 3.1.

**Pull request merged**